### PR TITLE
Improve test legends

### PR DESCRIPTION
**Pull request by pieer**
Improve test descriptions to be more understandable and to reflect exactly the nature of the test

### CODE REVIEW

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: User clicks on contribution link and is asked to add income details

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: tests language/person should be consistent. in our case, avoid using "you" and stick to the 3rd person.
> Opening income detail modal for the 1st time should only ask the user's income.

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: Good tests descriptions are hard to do and even harder to explain how to do them... The way I think about it is to write pragmatic test descriptions and focus on the final outcome of a task. For example, in this case, I would merge these 3 tests (244, 254 and 249), to reflect a user's "single real task".

> 'user1 fills their Income Details saying they are pledging 500$'.

That's the main goal in this chunk of tests. But keep the assertions for "edge cases" is totally fine (and desirable!), so you can keep the line asserting that introIncomeOrPledge doesn't exist with an inline comment to help other developers understand why we are "not.asserting" something.
```
cy.getByDT('introIncomeOrPledge').should('not.be.visible') // Make sure only the first question is visible
```

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: Same as before, I would merge the last 5 tests "User1 decides to switch income details to needing $100" with all the assertions (and inline comments) if needed.

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: > User1 adds a non...

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: > User2 sees the contributions they are receiving.
(should we assert the text? "{what} from {who}" text?

Comment (sandrina-p) [reply test/cypress/integration/group-proposals.spec.js]: User2 pledges $100 and sees the non-contributions they are receiving.

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: > user3 pledges $100 and sees who they are pledging to - $50 to user1 (Margarida)

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: > user4 and user2 change their pledges to $500 each. user1 sees the receiving contributions from 3 members.

**Review by sandrina-p — CHANGES_REQUESTED** [review]
I tried to explain the best I could how to write good tests descriptions, it's a constant developer challenge - naming and describing stuff.

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: typo, missing the s, removes*. Same goes for other test descriptions, edits, removes, gives, etc..

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: User1, without spaces, to match the username.

**Review by sandrina-p — CHANGES_REQUESTED** [review]
some last typos, besides that it's all fine!

**Review by taoeffect — DISMISSED** [review]
Approved, can be merged after @sandrina-p's latest review is addressed. Great job on this, thanks for improving the text! 👍

File: test/cypress/integration/group-proposals.spec.js
Comment (sandrina-p) [inline test/cypress/integration/group-proposals.spec.js]: @pieer I'm sorry to be this picky but there are still the same typos across the descriptions. users name without spaces (ex: user2 not user 2). incorrect verbs on 3rd person (receives, changes, gives, cancels, adds, etc...)

Comment (pieer) [reply test/cypress/integration/group-proposals.spec.js]: IT's actually not user2 but user2-234532 or user2-345423 so... not sure what's the point of your requests.. but sure I can change whatever you want so we can close this

Comment (sandrina-p) [reply test/cypress/integration/group-proposals.spec.js]: the ids change for reasons we all know, but the username itself (user1, user2, user3, etc...) are constant. Explicit saying the username in a description, help us to better understand their actions, the tests flow and it's easier to do a find/replace when needed.

**Review by sandrina-p — APPROVED** [review]

**Pull request merged**
